### ISSUE TITLE
jax-rs annotations defined in interface not working

### ISSUE THREAD

**Issue by neo-lestat** [micronaut-projects/micronaut-jaxrs#144]
### Expected Behavior

Hello, currently I have the jax-rs annotations in a interface and then this interface is implemented in a public class, I expected this endpoints are available (works in version 2.0.2)

### Actual Behaviour

The endpoint is not available with version 3.0.0

### Steps To Reproduce

This can be reproduce with: 

-  I defined an interface with the jaxrs annotations 

-  Add a class that implements this interface

```
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/greeter")
public interface GreeterService {
    @GET
    @Path("/string")
    @Produces(MediaType.TEXT_PLAIN)
    String greet();
}


public class GreeterImpl implements GreeterService {
    @Override
    public String greet() {
        return "hello!!!";
    }
}

```

### Version

3.0.0

**Comment by graemerocher** [comment]
`@Path` is not an inherited annotation (it is not meta-annotated with `@Inherited`) hence why it is not working. It is debatable if we should special case the JAX-RS annotations if they were not designed to be inherited.

**Comment by neo-lestat** [comment]
Hello @graemerocher , thank you for answer, I understand the situation. In my case there is a requirement to have this interfaces with the jaxrs annotations so it will be not possible to upgrade to micronaut 3.0.0. I understand that as special case maybe this will not add value.

**Comment by sdelamo** [comment]
@neo-lestat 

You could add `@Path` to implementer method: 

```java
public class GreeterImpl implements GreeterService {
    @Override
   `@Path`
    public String greet() {
        return "hello!!!";
    }
}
```

**Comment by neo-lestat** [comment]
Hello @sdelamo 

Thank you but my problem is that in the current poc by the moment is required to support this (due the jax-rs implementations that we already have). Personally i don't mine in add the annotations to the micronaut controllers

**Comment by lewisgcm** [comment]
@graemerocher JAX-RS annotations are designed to be inheritable. The spec [1] defines this behaviour as:
> JAX-RS annotations may be used on the methods and method parameters of a super-class or an implemented
interface. Such annotations are inherited by a corresponding sub-class or implementation class method
provided that the method and its parameters do not have any JAX-RS annotations of their own. Annotations
on a super-class take precedence over those on an implemented interface. The precedence over conflicting
annotations defined in multiple implemented interfaces is implementation specific. Note that inheritance of
class or interface annotations is not supported.
If a subclass or implementation method has any JAX-RS annotations then all of the annotations on the
superclass or interface method are ignored.

1 - https://download.oracle.com/otn-pub/jcp/jaxrs-2_0-fr-eval-spec/jsr339-jaxrs-2.0-final-spec.pdf?AuthParam=1654543366_76de17a8ba211b5f6186afe41ce9944d

**Comment by graemerocher** [comment]
Ok I regard this as a bug then. Bit weird they aren’t defined as such using a Meta annotation

**Comment by lewisgcm** [comment]
Yeah, it looks like there's an open issue on the jax-ws annotations repo to make those annotations Inherited. https://github.com/jakartaee/rest/issues/371.
